### ISSUE TITLE
Unknown unclosed entities not detected

### ISSUE THREAD

**Issue by ksmadsen** [petdance/html-lint#24]
Hi,

I've found another issue. Unknown unclosed entities are not detected by html_ok:

``` perl

perl -e 'use Test::HTML::Lint tests=>1; html_ok("known &unclosed &entities are not found");'
1..1                                                                                                                                                                                                                              
ok 1                                                                                                                                                                                                                              
```

The commit ksmadsen@c77ad836c67923f7791713815e1aa180523365b0 fixes this

However, it detects &oumlrhead as a single unknown entity, instead of an unclosed &ouml entity. The commit also fixes the tests that break because of this.
